tests/gis_tests/test_geoip.py

```
@@ -32,6 +33,16 @@ class GeoIPTest(unittest.TestCase):
     addr = '128.249.1.1'
     fqdn = 'tmc.edu'
 
+    def _is_dns_available(self, domain):
+        # Naive check to see if there is DNS available to use.
+        # Used to conditionally skip fqdn geoip checks.
+        # See #25407 for details.
+        try:
+            socket.gethostbyname(domain)
+            return True
+        except:
```

This seems fine for deprecated code, but could we please avoid a bare except?